bundles/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/SatelBinding.java

```diff
@@ -0,0 +1,235 @@
+import org.openhab.binding.satel.internal.event.NewStatesEvent;
+import org.openhab.binding.satel.internal.event.SatelEvent;
+import org.openhab.binding.satel.internal.protocol.Ethm1Module;
+import org.openhab.binding.satel.internal.protocol.IntRSModule;
+import org.openhab.binding.satel.internal.protocol.SatelMessage;
+import org.openhab.binding.satel.internal.protocol.SatelModule;
+import org.openhab.binding.satel.internal.protocol.command.NewStatesCommand;
+import org.openhab.binding.satel.internal.types.IntegraType;
+import org.openhab.core.binding.AbstractActiveBinding;
+import org.openhab.core.items.Item;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is main service class that helps exchanging data between OpenHAB and
+ * Satel module in both directions. Implements regular OpenHAB binding service.
+ * 
+ * @author Krzysztof Goworek
+ * @since 1.7.0
+ */
+public class SatelBinding extends AbstractActiveBinding<SatelBindingProvider> implements ManagedService, EventListener {
+
+	private static final Logger logger = LoggerFactory.getLogger(SatelBinding.class);
+
+	private long refreshInterval = 10000;
+	private String userCode;
+	private SatelModule satelModule = null;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected String getName() {
+		return "Satel Refresh Service";
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void execute() {
+		if (!isProperlyConfigured()) {
+			logger.warn("Binding not properly configured, exiting");
+			return;
+		}
+
+		if (!this.satelModule.isInitialized()) {
+			logger.debug("Module not initialized yet, skipping refresh");
+			return;
+		}
+
+		// for first refresh after connecting to the module, make full refresh
+		// after that check what has changed since last refresh
+		logger.trace("Sending 'get new states' command");
+		this.satelModule
+				.sendCommand(NewStatesCommand.buildMessage(this.satelModule.getIntegraType() == IntegraType.I256_PLUS));
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void updated(Dictionary<String, ?> config) throws ConfigurationException {
+		logger.trace("Binding configuration updated");
+
+		if (config == null)
+			return;
+
+		this.refreshInterval = getLongValue(config, "refresh", 10000);
+		this.userCode = (String) config.get("user_code");
+
+		int timeout = getIntValue(config, "timeout", 5000);
+		String host = (String) config.get("host");
+		if (StringUtils.isNotBlank(host)) {
+			this.satelModule = new Ethm1Module(host, getIntValue(config, "port", 7094), timeout,
+					(String) config.get("encryption_key"));
+		} else {
+			this.satelModule = new IntRSModule((String) config.get("port"), timeout);
+		}
```

Comment:
code consistency to me is a very important ingredient for readability and hence maintainability. In this case you mix up specialised methods and the direct handling in one method wich to me is inconsistent